targets/TARGET_NUVOTON/TARGET_M2354/TARGET_TFM/TARGET_NU_M2354/COMPONENT_TFM_S_FW/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2020-2021 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+include(${MBED_PATH}/targets/TARGET_NUVOTON/scripts/mbed_set_post_build_nuvoton.cmake)
```

Comment:
For Mbed CLI 2, we have provided a generic post build for all TF-M targets:
* [CMake definition](https://github.com/ARMmbed/mbed-os/blob/mbed-os-6.10.0/platform/FEATURE_EXPERIMENTAL_API/FEATURE_PSA/TARGET_TFM/TARGET_TFM_LATEST/scripts/mbed_set_post_build_tfm.cmake)
* [python script to run MCUboot imgtool](https://github.com/ARMmbed/mbed-os/blob/mbed-os-6.10.0/platform/FEATURE_EXPERIMENTAL_API/FEATURE_PSA/TARGET_TFM/TARGET_TFM_LATEST/scripts/generate_mbed_image.py)

To use it (e.g. Musca S1), simply do: https://github.com/ARMmbed/mbed-os/blob/9738b27c7df897c29e9769911d6794ba3e5b3f19/targets/TARGET_ARM_SSG/TARGET_MUSCA_S1/CMakeLists.txt#L4-L13

Is it possible to use it for M2354 too rather than having a dedicated one for this target?

Reply:
On M2354, we use a dedicated post-build script because:
1. The generic script supports only two-image signing (signing tf-m_s/mbed separately), but M2354 uses one-image signing (signing tf-m/mbed together).
1. In M2354 dedicated script, during signing, the padding option `--pad` is removed (so disable upgradeable by mcuboot) to decrease binary size for drag-n-drop flash. The having `--pad` version is still generated to enable future upgrade by mcuboot.
1. Difference of memory layout with ITS, NV counter, etc. can cause the generic script to not fit M2354.

Reply:
Thanks for the explanation. A separate script is fine now.

Maybe in the future we (Mbed OS) could consider making the generic script a bit more versatile - it should be quite doable given that the trusted-firmware-m project has a unified signing mechanism for all targets if I understand correctly. @ARMmbed/mbed-os-core FYI.